brax/physics/colliders.py

```diff
@@ -274,6 +274,119 @@ def apply(cap_a, cap_b, qp_a, qp_b):
 
     return P(vel_a + vel_b, ang_a + ang_b)
 
+class BoxHeightMap:
+  """ A collision between the box corners and a Heihtmap. """
+
+  def __init__(self, config: config_pb2.Config):
+    self.config = config
+    self.pairs = _find_body_pairs(config, 'box', 'heightMap')
+    if not self.pairs:
+      print("no_pairs")
+      return
+
+    body_idx = {b.name: i for i, b in enumerate(config.bodies)}
+    box_idxs = []
+    heightMap_idxs = []
+    corners = []
+    heights = []
+    sizes = []
+    meshSizes = []
+
+    for box, heightMap in self.pairs:
+      if not heightMap.frozen.all:
+        raise ValueError('active height maps unsupported: %s' % heightMap)
+      for i in range(8):
+        box_idxs.append(body_idx[box.name])
+        heightMap_idxs.append(body_idx[heightMap.name])
+        corner = jnp.array(
+            [i % 2 * 2 - 1, 2 * (i // 4) - 1, i // 2 % 2 * 2 - 1],
+            dtype=jnp.float32)
+        col = box.colliders[0]
+        corner = corner * vec_to_np(col.box.halfsize)
+        corner = math.rotate(corner, euler_to_quat(col.rotation))
+        corner = corner + vec_to_np(col.position)
+        corners.append(corner)
+
+        meshSize = heightMap.colliders[0].heightMap.meshSize
+        height = jnp.array(heightMap.colliders[0].heightMap.data).reshape((meshSize,meshSize))
+        heights.append(height)
+
+        sizes.append(heightMap.colliders[0].heightMap.size)
+        meshSizes.append(meshSize)
+
+    body = bodies.Body.from_config(config)
+    self.box = take(body, jnp.array(box_idxs))
+    self.heightMap = take(body, jnp.array(heightMap_idxs))
+    self.corner = jnp.array(corners)
+    self.size = jnp.array(sizes)
+    self.meshSize = jnp.array(meshSizes)
+    self.heights = jnp.array(heights)
+
+
+  def apply(self, qp: QP, dt: float) -> P:
+    """Returns impulse from a collision between box corners and a static height map.
+
+    Note that impulses returned by this function are *not* scaled by dt when
+    applied to parts.  Collision impulses are applied directly as velocity and
+    angular velocity updates.
+
+    Args:
+      qp: Coordinate/velocity frame of the bodies.
+      dt: Integration time step length.
+
+    Returns:
+      dP: Delta velocity to apply to the box bodies in the collision.
+      colliding: Mask for each body: 1 = colliding, 0 = not colliding.
+    """
+    if not self.pairs:
+      return P(jnp.zeros_like(qp.vel), jnp.zeros_like(qp.ang))
+
+    @jax.vmap
+    def apply(box, corner, qp_box, qp_heightMap, size, meshSize, heights):
+      pos, vel = math.to_world(qp_box, corner)
+      uv_pos = (pos[:2]-qp_heightMap.pos[:2])/size*(meshSize-1)
```

Comment:
Will this handle rotated height maps?  Users may want to use heightmaps as walls or ramps or some such. 
 Presumably, looking up the uv index of a rotated height map shouldn't be a problem, right?

Reply:
Indeed this does not handle rotated height maps. I am changing that.